### ISSUE TITLE
Loading VQE should be singled out as a separate script

### ISSUE THREAD

**Issue by marekgluza** [qiboteam/boostvqe#85]
The task of evaluating GCI on shots exemplifies a likely use case of our repo

- we have old vqe training
- we take some vqe unitary, we have some vqe circuit
- we want to boost it

Proposed workflow:
- [ ] add `gci+warmstart.py`
- [ ] the user loads a `qibo.Circuit` (or provides it somehow)
    - [ ] additionally the hamiltonian is provided
    - [ ] cost function is same game like loading vqe
- [ ] the script initializes https://github.com/qiboteam/boostvqe/blob/827b25fd1168a66f903c1820a4892da4d0121d57/src/boostvqe/utils.py#L220-L226
- [ ] It outputs GCI training results

This is proposing to have 2 scripts
- `main.py` trains vqe and boost with gci
-  `gci+warmstart.py` implements Fig 1 from Arxiv and boosts with gci after some `qibo.Circuit` wamstart

**Comment by Edoardo-Pedicillo** [comment]
I am not sure I have understood, but this could be solved in #92

**Comment by marekgluza** [comment]
Yes, part of this issue is solved in #92. For the journal submission it will be sufficient.

After we merge #92 I will readjust the status and switch to enhancement.

**Comment by marekgluza** [comment]
> * `gci+warmstart.py` implements Fig 1 from Arxiv and boosts with gci after some `qibo.Circuit` wamstart

To implement this, in `boost.py` the training would be conditional on a loading variable
https://github.com/qiboteam/boostvqe/blob/102b2abd697f919675073223e9f96b695425fa8d/src/boostvqe/boost.py#L179-L197
Our method is more general than just vqe so the user should either provide a warmstart circuit or will be running vqe as is done now.


> * `main.py` trains vqe and boost with gci

this is done by `boost.py` and will be sufficient for journal submission

**Comment by marekgluza** [comment]
Discussion with @Edoardo-Pedicillo :+1: 

- [ ] the user should be able to clone, and run 1 script which loads vqe (add vqe data to repo main) and obtain the plot as in current readme and main fig in paper
- [ ] refactor extras (?)
